File: 1. Fundamental-Level/CSharp/02. Java-Fundamentals-Octomber-2015/[HW] Java-Fundamentals/5. [HW] OOP-In-Java/Problem1_Geometry/PlaneShapes/Circle.java
package Problem1_Geometry.PlaneShapes;

import Problem1_Geometry.Vertices.Vertex2D;

public class Circle extends PlaneShape{
    private double radius;

    public Circle(Vertex2D vertex2D, double radius){
        super(vertex2D);
        this.radius = radius;
        this.perimeter = calculatePerimeter();
        this.area = calculateArea();
    }

    @Override
    public double calculateArea() {
        return Math.PI * this.radius * this.radius;
    }

    @Override
    public double calculatePerimeter() {
        return 2 * Math.PI * this.radius;
    }

    public double getRadius() {
        return radius;
    }

    public void setRadius(double radius) {
        this.radius = radius;
    }
}

File: 1. Fundamental-Level/CSharp/03. OOP-November-2015/[HW]/05. Encapsulation-Polymorphism/05. [HW]/01. Shapes/Figures/Circle.cs
namespace Shapes.Figures
{
    using System;
    using Interfaces;

    public class Circle : IShape
    {
        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public double Radius { get; private set; }

        public double CalculateArea()
        {
            return Math.PI * this.Radius * 2;
        }

        public double CalculatePerimeter()
        {
            return Math.PI * (this.Radius * this.Radius);
        }
    }
}